oauth2_provider/oauth2_backends.py

```diff
@@ -44,7 +45,26 @@ def _extract_params(self, request):
             del headers['wsgi.errors']
         if 'HTTP_AUTHORIZATION' in headers:
             headers['Authorization'] = headers['HTTP_AUTHORIZATION']
-        body = urlencode(request.POST.items())
+
+        return headers
+
+    def _extract_body(self, request):
+        """
+        Extracts the POST body from the Django request object
+        :param request: The current django.http.HttpRequest object
+        :return: provided POST parameters
+        """
+        return request.POST.items()
+
+    def _extract_params(self, request):
+        """
+        Extract parameters from the Django request object. Such parameters will then be passed to
+        OAuthLib to build its own Request object. The body should be encoded using OAuthLib urlencoded
+        """
+        uri = self._get_escaped_full_path(request)
+        http_method = request.method
+        headers = self._extract_headers(request)
+        body = urlencode(self._extract_body(request))
```

Comment:
+1

As we've discussed via hangout, it's better to promote ``extract_body`` and ``extract_headers`` as a public APIs